08-generic_cache/02_configured/cache.go

```diff
@@ -0,0 +1,48 @@
+package cache
+
+import "sync"
+
+// Cache is key, value storage.
+type Cache[K comparable, V any] struct {
+	data  map[K]V
+	mutex sync.Mutex
+}
+
+// New create a new Cache with an initialised data.
+func New[K comparable, V any]() Cache[K, V] {
+	return Cache[K, V]{
+		data: make(map[K]V),
+	}
+}
+
+// Read returns the associated value for a key,
+// or ErrNotFound if the key is absent.
+func (c *Cache[K, V]) Read(key K) (V, error) {
```

Comment:
This needs a locking as well. Also, this mistake should've come out from testing.